### PR TITLE
rename cloudName to name

### PR DESCRIPTION
**Pull request by car-roll**
cc @res0nance 
Preparing for https://github.com/jenkinsci/jenkins/pull/8310 where cloud name changes will be done on a separate page

The use of the variable name `cloudName` in `AmazonEC2Cloud` forces  jenkins core to have a special case of handling `cloudName` when it not a variable of `Cloud`.
https://github.com/jenkinsci/jenkins/blob/d41b0f5807923286ad33d4847af95e2c6c7c77ba/core/src/main/java/jenkins/agents/CloudSet.java#L229

Also changed `AmazonEC2CloudTest.testSshKeysCredentialsIdRemainsUnchangedAfterUpdatingOtherFields` to modify a different field.

### Testing done
manually created new `AmazonEC2Cloud`

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by car-roll** [comment]
Failing test fixed in https://github.com/jenkinsci/ec2-plugin/pull/875

File: src/test/resources/hudson/plugins/ec2/Ami.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ configuration-as-code:
 jenkins:
   clouds:
     - amazonEC2:
-        cloudName: "test"
+        name: "test"
```
Comment (basil) [inline src/test/resources/hudson/plugins/ec2/Ami.yml:7]: Similar issue also exists in the integration tests for `configuration-as-code-plugin`

**Comment by car-roll** [comment]
CI issues

**Comment by car-roll** [comment]
Closing in favor of a less invasive change

**Pull request closed without merging**